source/frontend/src/features/mapSideBar/acquisition/common/GenerateForm/hooks/useGenerateExpropriationForm1.tsx

```diff
@@ -37,18 +37,27 @@ export const useGenerateExpropriationForm1 = () => {
     }
     file.fileProperties = properties;
 
-    const ownerSolicitor = file.acquisitionFileInterestHolders?.find(
+    // fetch primary contact information for organizations within interest holders
+    if (interestHolders) {
+      await Promise.all(
+        interestHolders.map(async holder => {
```

Comment:
In parallel. We don't care about the order because they get assigned to the corresponding holder. We do wait for all of them to complete though. This pattern is already used in our code. See **GenerateFormContainer.tsx** - line 68-98

```tsx
    if (intHolders) {
      await Promise.all(
        intHolders.map(async holder => {
          if (holder.personId) {
            const person = (await getPersonConcept(holder?.personId))?.data;
            if (person) {
              generateRecipientsList.push(
                new LetterRecipientModel(
                  holder.interestHolderId!,
                  getInterestTypeString(holder.interestHolderType?.id ?? ''),
                  Api_GenerateOwner.fromApiPerson(person),
                  holder,
                ),
              );
            }
          } else if (holder.organizationId) {
            const org = (await getOrganizationConcept(holder?.organizationId))?.data;
            if (org) {
              generateRecipientsList.push(
                new LetterRecipientModel(
                  holder.interestHolderId!,
                  getInterestTypeString(holder.interestHolderType?.id ?? ''),
                  Api_GenerateOwner.fromApiOrganization(org),
                  holder,
                ),
              );
            }
          }
        }),
      );
    }
```

This link describes the parallel execution under "Waiting for finish" section - https://advancedweb.hu/how-to-use-async-functions-with-array-foreach-in-javascript/

Reply:
It's a bit subtle we do need ALL interest holders that are orgs with primary contacts to be updated. The Expropriation Form 1 requires that (also this is stated in the AC of the story). The owner solicitor happens to be part of the interest holders array but it is taken out and put separately in the Generate models (that was done prior to this PR)

Reply:
See my screenshot where it shows both the owner solicitor AND an interest holder with primary contacts